assets/src/stories-editor/components/editor-carousel/indicator.js

```diff
@@ -29,27 +36,100 @@ const Indicator = ( { pages, currentPage, onClick } ) => {
 	/* translators: %s: Page number */
 	const toolTip = ( pageNumber ) => sprintf( __( 'Go to page %s', 'amp' ), pageNumber );
 
+	const [ draggedPage, setDraggedPage ] = useState( null );
+	const { moveBlockToPosition } = useDispatch( 'core/block-editor' );
+	const { getBlockIndex } = useSelect( ( select ) => select( 'core/block-editor' ) );
+
 	return (
 		<ul className="amp-story-editor-carousel-item-list">
 			{ pages.map( ( page, index ) => {
 				const className = page.clientId === currentPage ? 'amp-story-editor-carousel-item amp-story-editor-carousel-item--active' : 'amp-story-editor-carousel-item';
+				const blockElementId = `amp-story-editor-carousel-item-${ page.clientId }`;
```

Comment:
`page.clientId` is used so often in this component, might be worthwhile to do `const { clientId } = page`